components/contribution-flow/ContributionBlocker.js

```diff
@@ -85,11 +71,48 @@ export const getContributionBlocker = (loggedInUser, account, tier, shouldHaveTi
     return { reason: CONTRIBUTION_BLOCKER.TIER_EXPIRED, type: 'warning', showOtherWaysToContribute: true };
   } else if (account.settings.disableCustomContributions && !tier) {
     return { reason: CONTRIBUTION_BLOCKER.NO_CUSTOM_CONTRIBUTION, type: 'warning', showOtherWaysToContribute: true };
+  } else if (
+    tier &&
+    tier.interval !== null &&
+    !account.host.supportedPaymentMethods.some(paymentMethod =>
+      ProvidersWithRecurringPaymentSupport.includes(paymentMethod),
+    ) &&
+    !isUserAdminOfCollectiveUnderSameHost(loggedInUser, account)
+  ) {
+    return {
+      reason: CONTRIBUTION_BLOCKER.NO_PAYMENT_PROVIDER,
+      type: 'warning',
+      showOtherWaysToContribute: true,
+      content: hasPaymentMethodAvailable(loggedInUser, account),
+    };
   } else {
     return null;
   }
 };
 
+const hasPaymentMethodAvailable = (loggedInUser, account) => {
```

Comment:
My [feedback](https://github.com/opencollective/opencollective-frontend/pull/6396#discussion_r638550559) on this one wasn't super clear, I was suggesting to add the **check** for this message in the helper - not the message, the goal being to group everything since we're displaying the same thing in both cases. Or said with code:

```es6

(...)

else if (!hasPaymentMethodAvailable(...)) {
  return {
    reason: CONTRIBUTION_BLOCKER.NO_PAYMENT_PROVIDER,
    ...
  }
}

(...)

const hasPaymentMethodAvailable = (...) => {
  return account.host.supportedPaymentMethods?.length &&
   !(
   tier &&
    tier.interval !== null &&
    !account.host.supportedPaymentMethods.some(paymentMethod =>
      ProvidersWithRecurringPaymentSupport.includes(paymentMethod),
    ) &&
    !isUserAdminOfCollectiveUnderSameHost(loggedInUser, account)
  )
}
```

Your approach is fine too, but if we go with it we should find a better name than `hasPaymentMethodAvailable` as one would expect a function starting with `has` to return a boolean.

Reply:
@Betree : Sorry for the misunderstanding. I've corrected it by renaming the method to `paymentMethodUnavailableWarning`. Somehow I though it was returning a boolean. 😉

Reply:
It seems that you forgot to commit this one :wink: Also, since `canContributeRecurring` covers this case, maybe we can also remove the `else if (!account.host.supportedPaymentMethods?.length) {` blocker condition above?

Reply:
@Betree : Ops sorry, my mistake; corrected both. 😉